File: problem22/python/problem22.py
def generateDictionary():
    dic = {}
    tmp = 'ABCDEFGHIJKLMNOPQRSTUVWXYZ'
    lis = []
    for i in tmp:
        lis.append(i)
    dic = dict([(i,j) for (i,j) in zip(lis,range(1,27))])
    return dic

def sumNamesScores():
    dic = generateDictionary()
    for i in open("p022_names.txt"):
        lis = sorted(i.replace('"',"").split(','))
        sumResult = 0
        for (i,j) in zip(lis,range(1,len(lis)+1)):
            tmp = 0
            for k in i:
                tmp += dic[k]
            sumResult += tmp * j
        return sumResult

print sumNamesScores()

File: problem22/Ruby/problem22.rb
def generateDict()
	dict = {}
	('A'..'Z').zip(1..26).each do |i,j|
		dict[i] = j
	end
	return dict
end
# 全局字典
$dict = generateDict()
def getRank(word)
	rank = 0
	word.each_char do |i|
		rank += $dict[i]
	end
	return rank
end

file = File.open('1.txt','r')
file.each do |i|
	rank = 0
	lis = i.gsub('"','').split(/,/).sort
	helper = 1
	lis.each do |j|
		rank += getRank(j) * helper
		helper = helper.succ
	end
	puts rank
end
file.close	
